internal/pkg/cli/deploy/workload.go

```diff
@@ -374,22 +388,85 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 	if err != nil {
 		return fmt.Errorf("login to image repository: %w", err)
 	}
+
+	var mu sync.Mutex
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+
+	// counter for indexing the labeled output buffers.
+	count := 0
```

Comment:
Isn't `count` too generic variable here? I am just saying because it is being used later down in the code and not around where it has been initialized.